de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/Env.java

```diff
@@ -1585,16 +1586,17 @@ public static void verifyLanguage(final Language language)
 			if (lang.equals(searchLangPart))
 			{
 				s_log.debug("Found similar Language {} for {}", AD_Language, language);
-				language.setAD_Language(AD_Language);
-				return;
+				final Language similarLanguage = language.toBuilder()
+						.AD_Language(AD_Language)
+						.build();
+				Language.addNewLanguage(similarLanguage);
+				return similarLanguage;
 			}
 		}
 
-		// No Language - set to Base Language
-		final String baseAD_Language = Language.getBaseAD_Language();
-		s_log.warn("Not System/Base Language={} - Set to Base Language: {}", language, baseAD_Language);
-		language.setAD_Language(baseAD_Language);
-	}   // verifyLanguage
+		// If the desired language (eg. en_US requested by the browser) is unavailable, we shall return the system base language.
+		return Language.getBaseLanguage();
```

Comment:
here why not doing the same as above?
i.e.
language.toBuilder().AD_Language(baseAD_Language).build()

Reply:
done